contents/security/passwordStorage.md

```diff
@@ -0,0 +1,98 @@
+
+As a developer, storing these usernames and passwords are essential to be able to refer to them later. While there are many ways to store these data - from text files to databases - this article will focus more on how to store passwords properly. For the purpose of this article, we will assume that you are storing the passwords in a database. Modern password mechanism typically involve **salting** and **hashing** the password. This article will describe what these mean and why they are important.
+
+### What's wrong with <trigger for="pop:plaintext">plaintext</trigger>?
+
+<popover id="pop:plaintext" title="_Plaintext_ refers to unencrypted information" placement="top">
+  <div slot="content">
+It is a cryptography term generally referring to text before encyption or after decrypting it. Another term for it is cleartext.
+  </div>
+</popover>
+
+While usernames are typically public information, thus encrypting them is not as important, passwords are typically secret information and require encryption. If passwords are stored in plaintext and your server's database is ever compromised, attackers can easily use the stored plaintext passwords to login as the user and potentially access other functionality. This grants the attacker more power than just reading values off the database and being unable to read them if they were encrypted.
+
+In addition,
+* Your employees can see the passwords of your users. If your employees are ever compromised, they might be willing to simply leak the password.
+* It is common for users to re-use the same password on multiple sites. If your passwords are stored as plaintext and accessed, attackers can easily re-use the same passwords on other websites (such as banking websites) and gain access to those.
+
+To prevent all of these measures, it is good to perform hashing on passwords. Unlike encryption, hashing is a <trigger for="pop:oneway">one-way function</trigger>. This means that an attacker cannot decrypt the encrypted password to retrieve the original password, thus ensuring that all of the above vulnerabilities isn't possible.
+
+<popover id="pop:oneway" title="A one way function is a function that is easy to compute in one way, but is close to impossible to compute the other way." placement="top">
+<div slot="content"></div>
+</popover>
+
+## What is hashing?
+
+Hashing is the act of transforming a set of data into another set of data that is difficult to reverse. As an example, suppose you wanted to multiply 2 prime numbers together, such as 17 and 11. This is easy for a computer to calculate, and it results in 187. However, suppose you were given a number that is the result of 2 prime numbers - such as 299. How do you find out which 2 prime numbers multipled together to create 299?
```

Comment:
I wonder if such a detailed example is necessary to get the idea of hashing across. I feel like it introduces a significant amount of complexity to the article, while not providing much additional value to the reader.

Reply:
Good point! I'll let others review/comment on it before deciding on what to do with it. Without the example I felt like it was hard to bring across the point of what hashing meant.

Reply:
Good idea!